www/js/components/download-track-page.js

```diff
@@ -1,89 +1,108 @@
+    this.props.dispatch(fn(id));
   }
 
   render() {
-    let downloaded = 0;
-    const tracks = Object.keys(this.state.tracks).map((trackId)=>{
-      let downloadButtonText = "Save";
-      let track = this.state.tracks[trackId];
-      downloaded += this.state.tracks[trackId].downloaded;
-      let progressBar = (
-        <ProgressBar progress={(track.downloaded/track.downloadSize) * 100}/>
-      )
-      if (track.downloading) {
-        downloadButtonText = "Cancel";
+    const { tracks } = this.props;
+
+    const downloaded = Object.keys(tracks).reduce((pre, cur) => {
+      return pre + (tracks[cur].status === 'fetched');
+    }, 0);
+
+    const rows = Object.keys(tracks).map(id => {
+      const track = tracks[id];
+
+      let progressBar;
+      if(track.isFetching === true) {
+        progressBar = (
+          <ProgressBar indeterminate={true} />
+        );
+      } else if(isFinite(track.isFetching)) {
+        progressBar = (
+          <ProgressBar progress={track.isFetching * 100} />
+        )
+      } else {
       }
-      if (track.downloaded >= track.downloadSize){
+
+      // TODO: Refactor this using O-O principles
+      let icon;
+      let downloadButtonText;
+      let isSave;
+      let action;
+      if(track.isFetching) {
+        downloadButtonText = 'Cancel';
+        isSave = false;
+        action = this.cancelTrack.bind(this, id);
+        icon = 'cloud_download';
+      } else if(track.status === 'available') {
         downloadButtonText = "Remove";
+        isSave = false;
+        action = this.removeTrack.bind(this, id);
+        icon = 'cloud_download';
+      } else {
+        downloadButtonText = "Save";
+        isSave = true;
+        action = this.saveTrack.bind(this, id, track.pkg);
+        icon = 'cloud_download';
       }
-      const isSave = downloadButtonText == "Save";
+
       return (
-        <div key={trackId}>
-          <div className="form-row">
-            <CardText> {track.name} </CardText>
-            <Button primary={isSave} accent={!isSave} raised
-                    onClick={
-                      this.trackDownload.bind(this,downloadButtonText,trackId)
-                    }>
-                {downloadButtonText} ({(track.downloadSize/1024).toFixed(0)} MB)
+        <Card key={id} className={'track-card'} shadow={3}>
+          <CardTitle>{track.name}</CardTitle>
+          <CardText>{track.description}</CardText>
+          <CardActions border={true}>
+            <Button primary={isSave} accent={!isSave} raised onClick={action}>
+              <span className='material-icons'>{icon}</span>
+              <span className='button-text'>{downloadButtonText}</span>
             </Button>
-          </div>
-          <div className="form-row">
-            { progressBar }
-          </div>
-        </div>
+            <span className='size-text'>{`${track.sizeMiB} MiB`}</span>
+
+            {/* TODO: Alternate activate controls to investigate */}
```

Comment:
possibly save this in a gist? we should not be leaving commented code in the repo because it may be useful.